TSPL.docc/ReferenceManual/Attributes.md

```diff
@@ -1866,7 +1866,87 @@ into code that calls the static methods of the result builder type:
     >> assert(builderOptional == manualOptional)
     ```
   -->
-- A code block or `do` statement
+- If the result builder implements
+  the `buildPartialBlock(first:)`
+  and `buildPartialBlock(accumulated:next:)` methods,
+  a code block or `do` statement becomes a call to those methods.
+  The first statement inside of the block
+  is transformed to become an argument
+  to the `buildPartialBlock(first:)` method,
+  and the remaining statements become nested calls
+  to the `buildPartialBlock(accumulated:next:)` method.
+  For example, the following declarations are equivalent:
```

Comment:
Should this section mention what happens when a builder implements both buildPartialBlock and buildBlock but with different availability?

Reply:
If partial block is less available than buildBlock the compiler would pick the latter.

Reply:
Is there a real-world use case for these semantics? (Apologies for my ignorance on this one- I spend 99% of my Swift time server-side these days; my only encounters with availability generally have to do with the compiler complaining that a given API requires a newer macOS as I'm trying to target Linux 😅.)

Reply:
Yeah, it's intended to cover a use-case where buildPartialBlock is introduced after buildBlock which gives it a way to support both based on the deployment target of the app in question (because i.e. buildBlock is limited in the number of elements it supports).

Reply:
We describe the case where you implement both "build partial block" and "build block" earlier in the chapter:

https://github.com/apple/swift-book/blob/main/TSPL.docc/ReferenceManual/Attributes.md?plain=1#L1501

@xedin Is that what you were thinking of?

Reply:
Yeah, that sounds good!